Tests/NIOTests/SocketOptionProviderTest.swift

```diff
@@ -58,15 +58,25 @@ final class SocketOptionProviderTest: XCTestCase {
         self.serverChannel = try? assertNoThrowWithValue(ServerBootstrap(group: group).bind(host: "127.0.0.1", port: 0).wait())
         self.clientChannel = try? assertNoThrowWithValue(ClientBootstrap(group: group).connect(to: serverChannel.localAddress!).wait())
 
+        // We need to join these multicast groups on the loopback interface to work around issues with rapidly joining and leaving
+        // many multicast groups. On some OSes, if we do that on a public interface, we can build up a kernel backlog of IGMP
+        // joins/leaves that may eventually lead to an ENOMEM and a spurious test failure. As joining/leaving groups on loopback
+        // interfaces does not require IGMP joins/leaves, forcing these joins onto the loopback interface saves us from this
+        // risk.
+        let v4LoopbackAddress = try! SocketAddress(ipAddress: "127.0.0.1", port: 0)
+        let v6LoopbackAddress = try! SocketAddress(ipAddress: "::1", port: 0)
```

Comment:
won't this blow up in non ipv6 systems?

Reply:
Nope. `inet_pton` works just fine on non-IPv6 systems, which is all this requires.